Sprint 10

ulia-lukash opened the pull request.

ulia-lukash committed:
Sprint 10

DanilaG commented on ImageFeed/Auth/Constants.swift:
```diff
@@ -0,0 +1,18 @@
+
+import Foundation
+
+let AccessKey = "kZGjT2dn25Tdqq24tFCBV-JxeZzVi4Xt8eIlaSGV9HU"
+let SecretKey = "IfvmVllIAcwAESfM423CxouBjXzyzRLgzIgZqDQymv4"
+let RedirectURI = "urn:ietf:wg:oauth:2.0:oob"
+
+let AccessScope = "public+read_user+write_likes"
+let DefaultBaseURL = URL(string: "https://api.unsplash.com")!
+let UnsplashAuthorizeURLString = "https://unsplash.com/oauth/authorize"
```
Давай назовём константы с маленькой буквы, чтобы они соответствовали остальному коду по стилю)

DanilaG commented on ImageFeed/Auth/Constants.swift:
```diff
@@ -0,0 +1,18 @@
+
+import Foundation
+
+let AccessKey = "kZGjT2dn25Tdqq24tFCBV-JxeZzVi4Xt8eIlaSGV9HU"
+let SecretKey = "IfvmVllIAcwAESfM423CxouBjXzyzRLgzIgZqDQymv4"
+let RedirectURI = "urn:ietf:wg:oauth:2.0:oob"
+
+let AccessScope = "public+read_user+write_likes"
+let DefaultBaseURL = URL(string: "https://api.unsplash.com")!
+let UnsplashAuthorizeURLString = "https://unsplash.com/oauth/authorize"
+
+let tokenBaseURL = "https://unsplash.com/oauth/token"
```
Для удобства в использовании автодополнения (и чтобы не запутаться) можно использовать name space для констант. Для этого, их можно обернуть в enum, а сами константы сделать static. Например, так:
```
enum Constants {
    static let accessKey = "kZGjT2dn25Tdqq24tFCBV-JxeZzVi4Xt8eIlaSGV9HU"
    ...
}
```
Тогда обращаться к значению можно следующим образом `Constants.accessKey`

DanilaG commented on ImageFeed/Auth/Constants.swift:
```diff
@@ -0,0 +1,18 @@
+let RedirectURI = "urn:ietf:wg:oauth:2.0:oob"
+
+let AccessScope = "public+read_user+write_likes"
+let DefaultBaseURL = URL(string: "https://api.unsplash.com")!
```
Давай избавимся от force unwrap в проекте, так как их использование может приводить к падению приложения?)

DanilaG commented on ImageFeed/Auth/OAuth2Service.swift:
```diff
@@ -0,0 +1,119 @@
+    }
+    
+    func fetchOAuthToken(
+            _ code: String,
+    completion: @escaping (Result<String, Error>) -> Void ){
```
Давай поправим отступы, чтобы код был более читаемым?)

ulia-lukash committed:
sprint 11

ulia-lukash committed:
Не получается получать URL аватарки на SplashScreen'е - вынесла в ProfileView.
Почему-то при создании TabBarItem'ов в коде те же самые картинки располагаются выше, чем, если бы было сделано через Storyboard.

ulia-lukash committed:
feat: ImageListService

ulia-lukash committed:
fix: nil token at photos fetch before login in

ulia-lukash committed:
fix: SingleImageView segue preparation

ulia-lukash committed:
Merge pull request #7 from ulia-lukash/sprint_12

Sprint 12

ulia-lukash committed:
Merge pull request #6 from ulia-lukash/sprint_11

Sprint 11
